editor/editor_node.cpp

```diff
@@ -2468,6 +2468,20 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 			OS::get_singleton()->set_window_fullscreen(!OS::get_singleton()->is_window_fullscreen());
 
 		} break;
+		case EDITOR_SCREENSHOT: {
+			Viewport *viewport = editor_interface->get_editor_viewport()->get_viewport();
+			viewport->set_clear_mode(Viewport::CLEAR_MODE_ONLY_NEXT_FRAME);
+			Ref<Image> img = viewport->get_texture()->get_data();
+			img->flip_y();
+			viewport->set_clear_mode(Viewport::CLEAR_MODE_ALWAYS);
+			Dictionary datetime = _OS::get_singleton()->get_datetime(false);
+			for (int32_t i = 0; i < datetime.size(); i++) {
+				datetime[datetime.get_key_at_index(i)] = String(datetime.get_value_at_index(i)).pad_zeros(2);
+			}
+			Error error = img->save_png(String("user://" + TTR("editor") + "_{year}_{month}_{day}_{hour}_{minute}_{second}.png")
+												.format(datetime));
```

Comment:
I would not add a line break here, the extra indentation makes it look worse than everything on a single line IMO.